backend/src/test/java/org/uresti/pozarreal/service/impl/PaymentConceptsServiceImplTests.java

```diff
@@ -0,0 +1,149 @@
+                .build();
+
+        org.uresti.pozarreal.model.PaymentConcept paymentConcept = org.uresti.pozarreal.model.PaymentConcept.builder()
+                .id("id2")
+                .label("label2")
+                .build();
+
+        Mockito.when(paymentConceptsRepository.save(argumentCaptor.capture())).thenReturn(paymentConcept);
+
+        //When:
+
+        PaymentConcept paymentConceptSaved = paymentConceptsService.save(paymentConceptDto);
+
+        //Then:
+        org.uresti.pozarreal.model.PaymentConcept parameter = argumentCaptor.getValue();
+
+        Assertions.assertThat(paymentConceptSaved).isNotNull();
+        Assertions.assertThat(parameter.getId()).isNotNull();
+        Assertions.assertThat(parameter.getId()).isNotEqualTo("id1");
+        Assertions.assertThat(parameter.getLabel()).isEqualTo("label1");
+        Assertions.assertThat(paymentConceptSaved.getId()).isEqualTo("id2");
+        Assertions.assertThat(paymentConceptSaved.getLabel()).isEqualTo("label2");
+
+        Mockito.verify(paymentConceptsRepository).save(parameter);
+        Mockito.verifyNoMoreInteractions(paymentConceptsRepository);
+    }
+
+    @Test
+    public void givenAnPaymentConcept_whenUpdate_thenPaymentConceptIsUpdated() {
+        //Given:
+        PaymentConceptsRepository paymentConceptsRepository = Mockito.mock(PaymentConceptsRepository.class);
+        PaymentConceptsServiceImpl paymentConceptsService = new PaymentConceptsServiceImpl(paymentConceptsRepository);
+
+        org.uresti.pozarreal.model.PaymentConcept paymentConcept = org.uresti.pozarreal.model.PaymentConcept.builder()
+                .id("id")
+                .label("label")
+                .build();
+
+        Mockito.when(paymentConceptsRepository.save(paymentConcept)).thenReturn(paymentConcept);
+
+        //When:
+        PaymentConcept paymentConceptSaved = paymentConceptsService.update(PaymentConceptMapper.entityToDto(paymentConcept));
+
+        //Then:
+        Assertions.assertThat(paymentConceptSaved.getId()).isEqualTo("id");
+        Assertions.assertThat(paymentConceptSaved.getLabel()).isEqualTo("label");
+    }
+
+    @Test
+    public void givenAnPaymentConceptWithPaymentConceptIdNull_whenUpdate_thenPaymentConceptThrowBadRequestDataException() {
+        //Given:
+        PaymentConceptsRepository paymentConceptsRepository = Mockito.mock(PaymentConceptsRepository.class);
+        PaymentConceptsServiceImpl paymentConceptsService = new PaymentConceptsServiceImpl(paymentConceptsRepository);
+
+        org.uresti.pozarreal.model.PaymentConcept paymentConcept = org.uresti.pozarreal.model.PaymentConcept.builder()
+                .label("label")
+                .build();
+
+        Mockito.when(paymentConceptsRepository.save(paymentConcept)).thenReturn(paymentConcept);
+
+        //When:
+        //Then:
+        Assertions.assertThatThrownBy(() -> paymentConceptsService.update(PaymentConceptMapper.entityToDto(paymentConcept)))
```

Comment:
same here.. remove the call to PaymentConceptMapper.entityToDto(paymentConcept) when calling to the tested method

Reply:
done